test/e2e/apply_run_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package e2e
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestApplyRun(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	// these key's values are random generated or k8s version based output. Will be used to remove these matching line from output and then match with expected output
+	keys := []string{"kapp.k14s.io/app", "creationTimestamp:", "resourceVersion:", "uid:", "selfLink:", "kapp.k14s.io/association"}
+	yaml := `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: simple-cm
+data:
+  hello_msg: good-morning-bangalore
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: simple-cm1
+data:
+  hello_msg: hello
+`
+	name := "test-apply-run"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+	cleanUp()
+	defer cleanUp()
+	logger.Section("creating an app with multiple resources", func() {
+		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "--apply-run"},
+			RunOpts{StdinReader: strings.NewReader(yaml)})
+		expectedOutput := `
+# add: configmap/simple-cm (v1) namespace: kapp-test
+---
+apiVersion: v1
+data:
+  hello_msg: good-morning-bangalore
+kind: ConfigMap
+metadata:
+  labels:
+  name: simple-cm
+  namespace: kapp-test
+# add: configmap/simple-cm1 (v1) namespace: kapp-test
+---
+apiVersion: v1
+data:
+  hello_msg: hello
+kind: ConfigMap
+metadata:
+  labels:
+  name: simple-cm1
+  namespace: kapp-test
+Succeeded
+`
+		out = strings.TrimSpace(replaceTarget(replaceSpaces(replaceTs(out))))
+		out = clearKeys(keys, out)
+
+		expectedOutput = strings.TrimSpace(replaceSpaces(expectedOutput))
+		require.Equal(t, out, expectedOutput, "output does not match1")
```

Comment:
Incorrect order doesn't matter much when the test are passing but could cause confusion in case the tests are failing. (Same in other places)
Also, match1 -> match
```suggestion
		require.Equal(t, expectedOutput, out, "output does not match")
```

Reply:
updated